### DISCUSSION TITLE
Use an existing table?

### DISCUSSION THREAD

**Discussion by RyanHow** [sensedeep/dynamodb-onetable#203] [Q&A] [answered]
This is more of a question than an issue...

Is there a way to use an existing table?

I wanted to keep the basic DynamoDB table managed directly by cloudformation (using CDK) and use OneTable for all the interactions with the table, handling GSIs, etc.

Is there a way to use `createTable()`, but pass in an existing minimal table?

The other way I can go about it is to use a custom resource and execute OneTable for the table creation and removal, then import the table for other resources to reference it. Would that be a better way of doing it?

Thanks!

**Comment by mobsense** [comment]
Sure, all you need is a schema and call the Table constructor. In fact, this is the normal mode of operation.

You don't (and shouldn't) call createTable for an existing table. createTable is really for dev and test time testing.

**Reply by RyanHow** [reply]
Ok great. Thanks!

So if I'm not using createTable, would I handle GSIs in migrations to match the schema using the updateTable() function?

(I was assuming that createTable would create the table and the GSIs)

**Comment by mobsense** [comment, marked as answer]
You don't call updateTable either. create/delete/update table are for making actual changes to the table keys.

If you want to create or delete GSIs, then use updateTable. and yes, you could do that in a migration.

For normal operation, OneTable reads the keys from the table to determine the key structure. Also, your schema should include the keys in the {indexes} section. These should (must) match the actual keys.

**Reply by RyanHow** [reply]
Thanks for clarifying!

Essentially what I am needing to do is to sync up the schema and the actual table. So I think updateTable is what I need to do in an initial migration.

Thanks for the swift and helpful response! I'm getting excited about using DynamoDb again after finding this library.
